policy/modules/contrib/frr.te

```diff
@@ -0,0 +1,81 @@
+policy_module(frr, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type frr_t;
+type frr_exec_t;
+init_daemon_domain(frr_t, frr_exec_t)
+
+type frr_log_t;
+logging_log_file(frr_log_t)
+
+type frr_conf_t;
+files_config_file(frr_conf_t)
+
+type frr_unit_file_t;
+systemd_unit_file(frr_unit_file_t)
+
+type frr_var_run_t;
+files_pid_file(frr_var_run_t)
+
+########################################
+#
+# frr local policy
+#
+allow frr_t self:capability { dac_override kill net_bind_service net_raw setgid setuid };
+allow frr_t self:process setcap;
+allow frr_t self:netlink_route_socket rw_netlink_socket_perms;
+allow frr_t self:tcp_socket { connect connected_stream_socket_perms };
+allow frr_t self:udp_socket create_socket_perms;
+allow frr_t self:rawip_socket create_socket_perms;
+
+allow frr_t frr_conf_t:dir list_dir_perms;
+read_files_pattern(frr_t, frr_conf_t, frr_conf_t)
+read_lnk_files_pattern(frr_t, frr_conf_t, frr_conf_t)
+
+manage_dirs_pattern(frr_t, frr_log_t, frr_log_t)
+manage_files_pattern(frr_t, frr_log_t, frr_log_t)
+manage_lnk_files_pattern(frr_t, frr_log_t, frr_log_t)
+logging_log_filetrans(frr_t, frr_log_t, { dir file lnk_file })
+
+manage_dirs_pattern(frr_t, frr_var_run_t, frr_var_run_t)
+manage_files_pattern(frr_t, frr_var_run_t, frr_var_run_t)
+manage_lnk_files_pattern(frr_t, frr_var_run_t, frr_var_run_t)
+manage_sock_files_pattern(frr_t, frr_var_run_t, frr_var_run_t)
+files_pid_filetrans(frr_t, frr_var_run_t, { dir file lnk_file })
+
+allow frr_t frr_exec_t:dir search_dir_perms;
+read_lnk_files_pattern(frr_t, frr_exec_t, frr_exec_t)
+can_exec(frr_t, frr_exec_t)
+
+kernel_read_system_state(frr_t)
+
+corecmd_exec_bin(frr_t)
+
+corenet_tcp_bind_bgp_port(frr_t)
+corenet_tcp_bind_zebra_port(frr_t)
+corenet_udp_bind_router_port(frr_t)
+
+auth_use_nsswitch(frr_t)
+
+domain_use_interactive_fds(frr_t)
+
+files_manage_generic_locks(frr_t)
```

Comment:
What exact lock file is frr trying to access? If it's its own lock file, then perhaps it should be labeled as `frr_lock_t`, avoiding the need for this generic rule?

Reply:
Frr trying access to lock files like **/var/lock/susbys/bgpd**. This lock files are related with protocols, which frr use. I think, that we are not able to label dir /varl/lock/subsys as frr_lock_t, because dir is not use only for frr lock files, but It's possible to label each lock file which use frr or what do you think?

Reply:
I tried to label each lock file and test case with these changes was succesfull without any fail and AVC's, so I update this PR.

Reply:
The `files_manage_generic_locks(frr_t)` is still there in the last version - can you try with it removed?

Reply:
Oh sorry, I forgot to delete it in PR version. Now It should be everything updated.